src/code/main.c

```diff
@@ -90,7 +90,7 @@ void Main(void* arg) {
     StackCheck_Init(&sPadMgrStackInfo, sPadMgrStack, STACK_TOP(sPadMgrStack), 0, 0x100, "padmgr");
     PadMgr_Init(&gPadMgr, &sSerialEventQueue, &gIrqMgr, THREAD_ID_PADMGR, THREAD_PRI_PADMGR, STACK_TOP(sPadMgrStack));
 
-    AudioMgr_Unlock(&gAudioMgr);
+    AudioMgr_WaitForInit(&gAudioMgr);
```

Comment:
Any idea why this is, iirc the main thread has lowest priority so the audiomgr thread will have already run up to the point where it's blocking on something basically before AudioMgr_Init returns

Reply:
ah I got confused between the "main" thread and the "graph" thread, thanks

even with main boosting its priority to 15=audiomgr+3, in the mean time the bulk of execution happens in the graph thread which is audiomgr-1 in priority, so it's another reason this is useless

but oh well :shrug: explicit synchronization doesn't hurt